src/lambdaconnect_model/scope_single_tag.clj

```diff
@@ -0,0 +1,86 @@
+(ns lambdaconnect-model.scope-single-tag
+  (:require [clojure.edn] 
+            [clojure.set :as set]
+            [lambdaconnect-model.utils :refer [relevant-tags]]))
+
+(defn- build-dependency-tree
+  "Reads scoping and builds tree which describes dependency between tags
+   each edge represents a dependency: 
+   if tag B has ingoing edge from tag A it means that tag A is present in tag's B contraint
+   if tag A has outgoing edge to B it means that tag B has tag A in its contraints
+   edges are represent as map of sets: tag #{set of in/out going edges}, roots are stored in set
+   returns a tuple [ingoding edges, outgoing edges, roots (where root is a tag with :user/uuid in its contraints)]" 
+  [scoping]
+  (let [tags (keys scoping)
+        empty-edge-map (->> tags
+                            (map (fn [tag] [tag #{}]))
+                            (into {})) 
+        dependency-tree (reduce (fn [[in out roots] [tag attrs]]
+                                      (let [tag-parents (relevant-tags (:constraint attrs))
+                                            root? (contains? tag-parents :user)
```

Comment:
może zdefiniować jakąś zmienną 
```
(def root-tag :user)
```
bo tak to nagle pojawia się to `:user` i jest to trochę mało czytelne

Reply:
:user -> root-tag